### PR TITLE
Implement the rest of Locomotor

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/logic/object/locomotor.cpp [hunk 1]
```diff
@@ -1853,3 +1908,295 @@ void Locomotor::Start_Move()
...
+    }
+
+    m_moveFrame = g_theGameLogic->Get_Frame() + 2.5f * 30.0f;
+    Set_Flag(IS_BRAKING, false);
+    PhysicsBehavior *physics = obj->Get_Physics();
+
+    if (physics == nullptr) {
+        captainslog_dbgassert(false, "you can only apply Locomotors to objects with Physics");
+        return true;
+    } else {
```
Comment (tomsons26) [inline src/game/logic/object/locomotor.cpp]: redundant

File: src/game/logic/object/locomotor.cpp [hunk 2]
```diff
@@ -1853,3 +1908,295 @@ void Locomotor::Start_Move()
...
 }
+
+bool Locomotor::Loco_Update_Maintain_Current_Position(Object *obj)
+{
+    if (!Get_Flag(MAINTAIN_POS_IS_VALID)) {
+        m_maintainPos = *obj->Get_Position();
+        Set_Flag(MAINTAIN_POS_IS_VALID, true);
+    }
+
+    m_moveFrame = g_theGameLogic->Get_Frame() + 2.5f * 30.0f;
```
Comment (tomsons26) [inline src/game/logic/object/locomotor.cpp:1919]: is this same as startmove?

File: src/game/logic/object/locomotor.cpp (hunk 1 above)
Comment (jonwil) [reply src/game/logic/object/locomotor.cpp]: fixed.

File: src/game/logic/object/locomotor.cpp (hunk 2 above)
Comment (jonwil) [reply src/game/logic/object/locomotor.cpp:1919]: I don't know but I can confirm its 100% not an inlined copy of Start_Move.

File: src/game/logic/object/locomotor.cpp [hunk 3]
```diff
@@ -1853,3 +1908,295 @@ void Locomotor::Start_Move()
...
+            Utf8String name;
+            xfer->xferAsciiString(&name);
+            LocomotorTemplate *lt = g_theLocomotorStore->Find_Locomotor_Template(Name_To_Key(name));
+
+            if (!lt) {
+                captainslog_debug("LocomotorSet::xfer - template %s not found", name.Str());
+                throw XFER_STATUS_NOT_FOUND;
+            }
+
+            Locomotor *l = new Locomotor(lt);
```
Comment (tomsons26) [inline src/game/logic/object/locomotor.cpp]: this should be calling New_Locomotor of the loco store

Comment (jonwil) [reply src/game/logic/object/locomotor.cpp]: fixed.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**